src/vorta/profile_export.py

```diff
@@ -130,6 +130,9 @@ def to_db(self, overwrite_profile=False, overwrite_settings=True):
         # Set the profile ids to be match new profile
         for source in self._profile_dict['SourceFileModel']:
             source['profile'] = self.id
+        # Delete existing Sources to avoid duplicates
+        for source in SourceFileModel.select().where(SourceFileModel.profile == self.id):
```

Comment:
You don't need a loop here. Can be done with one query. E.g.

https://stackoverflow.com/a/33314956